python/tvm/relay/tensorrt.py

```diff
@@ -0,0 +1,175 @@
+        return super().visit_call(expr)
+
+class RemoveRedundantTranspose(ExprMutator):
+    """
+    Removes Transpose([1, 0]) followed by Transpose([1, 0]). This pass, when
+    preceded by with RemoveMultiplyByOne is intended to remove a pattern of
+    Transpose([1, 0]) -> Scale(1.0f) -> Transpose([1, 0]) produced by
+    PyTorch's addmm operator.
+    """
+    def check_axes(self, axes):
+        return len(axes) == 2 and int(axes[0].value) == 1 and int(axes[1].value) == 0
+
+    def visit_call(self, expr):
+        if expr.op == tvm.relay.op.get("transpose"):
+            if self.check_axes(expr.attrs['axes']):
+                if isinstance(expr.args[0], Call) and \
+                   expr.args[0].op == tvm.relay.op.get("transpose"):
+                    if self.check_axes(expr.args[0].attrs['axes']):
+                        return expr.args[0].args[0]
+        return super().visit_call(expr)
+
+def GetTrtVersion():
+    """Gets the version of TensorRT that TVM is built against.
+
+    Returns
+    -------
+    ret: Tuple[int]
+        TensorRT version as a tuple of major, minor, and patch number. If TVM
+        is not built with TensorRT, an empty tuple is returned instead.
+    """
+    return tuple(map(int, _transform.GetTrtVersion()))
+
+def EnableTrt(mod, params=None, trt_version=None):
+    """Converts the entire relay program into one that can be executed using
+    TensorRT. If any of the operators are not supported by the TensorRT
+    conversion, the unmodified program will be returned instead.
+
+    Parameters
+    ----------
+    mod: Module
+        The original module.
+
+    params : dict of str to NDArray
+        Input parameters to the graph that do not change
+        during inference time. Used for constant folding.
+
+    trt_version : Optional[Tuple[int]]
+        Which version of TensorRT to target for partitioning as a tuple of
+        (major, minor, patch). If not specified, will attempt to get using
+        GetTrtVersion.
+
+    Returns
+    -------
+    ret: tvm.relay.Pass
+        The registered pass that partitions the Relay program.
+    """
+    if not trt_version:
+        trt_version = GetTrtVersion()
+        # If TVM wasn't built against TRT, default to target TRT 6. Since the
+        # actual conversion to TRT is done at runtime, building against TRT is
+        # not required for compilation.
+        if not trt_version:
+            trt_version = (6, 0, 1)
+    assert isinstance(trt_version, (list, tuple))
+    assert len(trt_version) == 3
+
+    # Apply passes required for TRT
+    mod = relay.transform.RemoveUnusedFunctions()(mod)
+    mod = relay.transform.InferType()(mod)
+    mod = relay.transform.ConvertLayout('NCHW')(mod)
+    mod['main'] = ConvertLayoutTransform().visit(mod['main'])
+    mod['main'] = RemoveDropout().visit(mod['main'])
+    mod['main'] = RemoveMultiplyByOne().visit(mod['main'])
+    mod['main'] = RemoveRedundantTranspose().visit(mod['main'])
```

Comment:
These only assume there is only on function which is "main" in the module or other functions in the module could be ignored. Please document it if this is a valid assumption.

Reply:
The latter three passes can be implemented into one and named as "TensorPreprocess", right?

Reply:
Thanks, I added a comment about "main". I don't think the passes can be combined since RemoveMultiplyByOne() must run before RemoveRedundantTranspose().

Reply:
I think Zhi's suggestion is to add a new method - TRTPreProcess and move all the functions there